### PR TITLE
Remove data directory only if replica creation failed

### PR DESCRIPTION
**Pull request by CyberDem0n**
And follow the right node after replica creation (it was following the same node from which it took the backup).

Fixes https://github.com/zalando/patroni/issues/190

### CODE REVIEW

File: patroni/ha.py [hunk 1]
```diff
@@ -67,11 +67,14 @@ def touch_member(self):
         self.dcs.touch_member(json.dumps(data, separators=(',', ':')))
 
     def clone(self, clone_member=None, msg='(without leader)'):
-        if self.state_handler.bootstrap(cluster_initialized=True, clone_member=clone_member):
+        if self.state_handler.clone(clone_member):
             logger.info('bootstrapped %s', msg)
+            node_to_follow = self._get_node_to_follow(self.dcs.get_cluster())
+            if node_to_follow and node_to_follow.name == self.state_handler.name:
```
Comment (alexeyklyukin) [inline patroni/ha.py]: Should we do the check for node_to_follow.name == self.state_handler.name inside the _get_node_to_follow and just return None in this case?

File: patroni/postgresql.py [hunk 2]
```diff
@@ -126,7 +126,12 @@ def get_local_address(self):
         return local_address + ':' + self.port
 
     def get_postgres_role_from_data_directory(self):
```
Comment (alexeyklyukin) [inline patroni/postgresql.py]: This function is called from the set_role, so it should return only those roles accepted by set_role (which is not 'uninitialized').

Comment (CyberDem0n) [inline patroni/postgresql.py]: `set_role` doesn't apply any restrictions on value

File: patroni/ha.py (hunk 1 above)
Comment (CyberDem0n) [inline patroni/ha.py]: fixed

File: patroni/postgresql.py [hunk 3]
```diff
@@ -693,41 +700,32 @@ def sync_replication_slots(self, cluster):
...
+        ret = self.create_replica(clone_member) == 0
+        if ret:
+            self.delete_trigger_file()
+            self.restore_configuration_files()
+        return ret
 
+    def bootstrap(self):
+        """
             Failure during initdb always leads to an exception, since there is
             no point in continuing if initdb fails. For the rest of the cases,
```
Comment (alexeyklyukin) [inline patroni/postgresql.py]: This comment is outdated, since this function is not returning False anymore

**Comment by alexeyklyukin** [comment]
+1

**Pull request merged**